caffe2/operators/distance_op.cc

```diff
@@ -223,10 +223,27 @@ bool DotProductOp<float, CPUContext>::RunOnDevice() {
   return true;
 }
 
+vector<TensorShape> TensorInferenceForDotProduct(
+    const OperatorDef& /* def */,
+    const vector<TensorShape>& in) {
+  vector<TIndex> dims;
+  if (in[0].dims().size() > 0) {
+    dims.push_back(in[0].dims(0));
```

Comment:
looks like shape is going to be on size 1 max. Are you sure it is correct?

Reply:
According to the documentation it's required to be 1-D (https://caffe2.ai/docs/operators-catalogue.html#dotproduct).  That said, the documentation also states that X and Y inputs are only allowed to be  1-D or 2-D, yet the implementation appears to support inputs 0- to N-D.  The inference here sets the output to 0-D if input X is 0-D, otherwise 1-D.

Reply:
could you make sure that this is in sync with implementation then and enforce that this is a case?